probe-rs/src/session.rs

```diff
@@ -67,7 +83,7 @@ impl<'a> AsMut<dyn DebugProbe + 'a> for ArchitectureInterface {
 }
 
 impl Session {
-    /// Open a new session with a given debug target
+    /// Open a new session with a given debug target.
     pub fn new(
```

Comment:
Do we even require this method to be public? I think the intended way is to go from `Probe::attach()` and `Session::auto_attach()`? I wouldn't mind leaving this public, but since you changed the visibility of other methods too, we could change this too :)

Reply:
I agree that it probably shouldn't be public, I will it make it `pub(crate)`.